azure_pipeline-global.yaml

```diff
@@ -0,0 +1,335 @@
+# Docs:
+# https://aka.ms/yaml
+
+trigger:
+  batch: true
+  branches:
+    include:
+    - master
+  paths:
+    include:
+    - /modules/azure-landing-zone/*
+    - /components/global/*
+    - /environments/demo/demo.tfvars
+    - /environments/dev/dev.tfvars
+    - /environments/ithc/ithc.tfvars
+    - /environments/ldata/ldata.tfvars
+    - /environments/prod/prod.tfvars
+    - /environments/ptl/ptl.tfvars
+    - /environments/sbox/sbox.tfvars
+    - /environments/stg/stg.tfvars
+    - /environments/test/test.tfvars
+    
+variables:
+  terraformVersion: 0.12.26
+  agentPool: 'hmcts-agent-pool'
+  action:
+
+stages:
+  - stage: Precheck
+    jobs:
+      - job: Validate
+        pool: ${{ variables.agentPool }}
+        steps:
+        - task: AzureKeyVault@1
+          displayName: Retrieve keyvault secret for ADO token
+          inputs:
+            ConnectedServiceName: azurerm-sandbox
+            keyVaultName: infra-vault-nonprod
+            secretsFilter: 'azure-devops-token'
+            runAsPreJob: false
+        - task: Bash@3
+          displayName: Prevent parallel run
+          env:
+            thisbuild: $(Build.BuildId)
+            pipelinedefinition: $(System.DefinitionId)
+          inputs:
+            targetType: 'inline'
+            script: | 
```

Comment:
this should probably be extracted out to a script? possibly even a library?

Reply:
Latest merge has included scriptPath for both builds-checks & UnlockTfState-check